contracts/Presale.sol

```diff
@@ -0,0 +1,289 @@
+            account: _sender,
+            adminAllocation: false
+        });
+
+        allocation = _sync(_purchaseConfig);
+    }
+
+    function allocate(address account, uint256 amountUSD) external override onlyOwner returns (uint256 allocation) {
+        PurchaseConfig memory _purchaseConfig = PurchaseConfig({
+            asset: address(0),
+            amountAsset: amountUSD,
+            amountUSD: amountUSD,
+            account: account,
+            adminAllocation: true
+        });
+
+        allocation = _sync(_purchaseConfig);
+    }
+
+    receive() external payable {
+        purchase();
+    }
+
+    function _sync(PurchaseConfig memory purchaseConfig) private returns (uint256) {
+        PresaleConfig memory _config = $config;
+        RoundConfig[] memory _rounds = $rounds;
+
+        require(block.timestamp >= _config.startDate, "RAISE_NOT_STARTED");
+        require(block.timestamp <= _config.endDate, "RAISE_ENDED");
+        require(
+            purchaseConfig.amountUSD >= _config.minDepositAmount || _config.minDepositAmount == 0, "MIN_DEPOSIT_AMOUNT"
+        );
+
+        uint256 _totalAllocation;
+        uint256 _totalPurchaseAmountAsset;
+
+        uint256 _remainingUSD = purchaseConfig.amountUSD;
+        uint256 _userAllocationRemaining = _config.maxUserAllocation - $userTokensAllocated[purchaseConfig.account];
+
+        uint256 i = $currentRoundIndex;
+        for (i; i < _rounds.length && _remainingUSD > 0 && _userAllocationRemaining > 0; ++i) {
+            RoundConfig memory _round = _rounds[i];
+            uint256 _roundAllocated = $roundAllocated[i];
+            uint256 _roundAllocationRemaining =
+                _roundAllocated < _round.tokensAllocated ? _round.tokensAllocated - _roundAllocated : 0;
+
+            if (_roundAllocationRemaining == 0) continue;
+
+            uint256 _roundAllocation = _remainingUSD * PRECISION / _round.tokenPrice;
+
+            if (_roundAllocation > _roundAllocationRemaining) {
+                _roundAllocation = _roundAllocationRemaining;
+            }
+            if (_roundAllocation > _userAllocationRemaining) {
+                _roundAllocation = _userAllocationRemaining;
+            }
+
+            require(_roundAllocation > 0, "MIN_ALLOCATION");
+
+            uint256 _tokensCostUSD = _roundAllocation * _round.tokenPrice / PRECISION;
+            _remainingUSD -= _tokensCostUSD;
+
+            _userAllocationRemaining -= _roundAllocation;
+            _totalAllocation += _roundAllocation;
+
+            uint256 _roundPurchaseAmountAsset = _tokensCostUSD * purchaseConfig.amountAsset / purchaseConfig.amountUSD;
+            _totalPurchaseAmountAsset += _roundPurchaseAmountAsset;
+
+            PurchaseConfig memory _roundPurchaseConfig;
+            _roundPurchaseConfig.asset = purchaseConfig.asset;
+            _roundPurchaseConfig.amountAsset = _roundPurchaseAmountAsset;
+            _roundPurchaseConfig.amountUSD = _tokensCostUSD;
```

Comment:
because we are doing this. 
we might as well just delete the _deposit method and bring the logic here so we dont need to do this.